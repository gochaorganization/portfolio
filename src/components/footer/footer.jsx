import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import _ from 'lodash'

import ListLink from './listlink'

import './footer.sass'

export class Footer extends Component {
  renderLink = (link, key) => {
    return <ListLink key={key} link={link} />
  }

  render() {
    const { footerLinks } = this.props
    return (
      <footer className="footer">
        <ul className="footer__list">
          {_.map(footerLinks, (item, key) => this.renderLink(item, key))}
        </ul>

        <div className="footer__credentials">
          Made by <span className="footer__credentials--danger">Selleo</span>
        </div>
      </footer>
    )
  }
}

Footer.propTypes = {
  footerLinks: PropTypes.array.isRequired,
}

const mapStateToProps = ({ portfolio: { footerLinks } }) => ({
  footerLinks,
})

export default connect(mapStateToProps)(Footer)
