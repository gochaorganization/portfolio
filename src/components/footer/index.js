/* eslint-disable import/no-named-as-default */
import Footer from './footer'

export default Footer
