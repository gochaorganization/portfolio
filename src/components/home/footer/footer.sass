@import '../../../application'

.footer
    display: flex    
    flex-direction: column
    justify-content: flex-start

    &__list
        display: flex
        flex-direction: column
        list-style: none
        margin: 0
        padding: 0 0 $m-base

        &-link 
            align-items: center
            color: $dark-grey
            display: flex
            font-size: $font-xs
            margin: $m-xs 0   

            &:before
                display: inline-block
                margin-right: $m-sm
                width: $icon-size
        
    &__credentials
        color: $grey
        font-size: $font-xs

        &--danger
            color: $danger

+media-breakpoint-up(sm)
    .footer
        flex-direction: row
        justify-content: space-between
        
        &__list        
            flex-direction: row
            padding: 0 

            &-link 
                margin: 0 $m-xs       
            
        &__credentials
            align-self: center
