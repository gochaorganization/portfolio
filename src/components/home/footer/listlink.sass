@import '../../../application'

.list-link 
    align-items: center
    color: $grey
    display: flex
    font-size: $font-xs
    margin: $m-xs 0   

    &:before
        display: inline-block
        margin-right: $m-sm
        width: $icon-size

+media-breakpoint-up(sm)
    .list-link 
        margin: 0 $m-xs       
