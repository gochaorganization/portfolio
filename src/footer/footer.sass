@import '../application'

.icon--facebook
    +icon(test, 'ea91')

.footer
    display: flex    
    flex-direction: column
    justify-content: flex-start
    +media-breakpoint-up(sm)
        flex-direction: row
        justify-content: space-between

    &__list
        display: flex
        flex-direction: column
        margin: 0
        padding: $m-base 0
        list-style: none
        +media-breakpoint-up(sm)
            flex-direction: row

        &-link 
            display: block
            margin: $m-xs 0
            color: $grey
            font-size: $font-xs
            +media-breakpoint-up(sm)
            margin: 0 $m-xs       

    &__credentials
        color: $grey
        font-size: $font-xs
        +media-breakpoint-up(sm)
            align-self: center

        &--red
            color: $red
        
