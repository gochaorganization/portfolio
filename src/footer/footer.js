import React from "react";
import "./footer.sass";

const Footer = () => <div />;

export default Footer;
