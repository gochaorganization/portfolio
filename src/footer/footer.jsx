import React, { Component } from "react";
import _ from "lodash";

import ListLink from "./listlink";

import "./footer.sass";

class Footer extends Component {
  state = {
    links: [
      { name: "twitter", link: "https://twitter.com" },
      { name: "dribble", link: "https://dribbble.com/" },
      { name: "facebook", link: "https://facebook.com" }
    ]
  };

  renderLink = (link, key) => {
    return <ListLink key={key} link={link} />;
  };

  render() {
    const { links } = this.state;
    return (
      <footer className="footer">
        <ul className="footer__list">
          {_.map(links, (item, key) => this.renderLink(item, key))}
        </ul>

        <div className="footer__credentials">
          Made by <span className="footer__credentials--danger">Selleo</span>
        </div>
      </footer>
    );
  }
}

export default Footer;
