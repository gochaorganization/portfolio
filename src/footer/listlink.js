import React from "react";
import _ from "lodash";
import classNames from "classnames";

import "./listlink.sass";

const ListLink = ({ link: { name, address } }) => (
  <li>
    <a
      className={classNames("list-link", {
        [`icon-${name}`]: name
      })}
      href={address}
    >
      {_.upperFirst(name)}
    </a>
  </li>
);

export default ListLink;
