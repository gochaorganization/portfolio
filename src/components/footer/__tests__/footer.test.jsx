import React from 'react'
import { shallow, mount } from 'enzyme'
import { Footer } from '../footer'
import ListLink from '../listlink'

describe('Footer tests', () => {
  const props = {
    footerLinks: [
      { name: 'twitter', address: 'https://twitter.com' },
      { name: 'dribble', address: 'https://dribbble.com/' },
      { name: 'facebook', address: 'https://facebook.com' },
    ],
  }

  test('renders correctly', () => {
    const footer = shallow(<Footer {...props} />)
    expect(footer).toMatchSnapshot()
  })

  test('menu should render links', () => {
    const footer = shallow(<Footer {...props} />)
    expect(footer.find(ListLink)).toHaveLength(3)
  })

  test('menu link renders correctly', () => {
    const footer = mount(<Footer {...props} />)

    
    expect(
      footer
        .find(ListLink)
        .at(0)
        .text()
    ).toEqual('Twitter')
  })
})

/*
1. Snapshot
2. Testing props. 
3. Testing data types (optional) - jest-extended 
4. Event Testing
5. Testing state

6.Check test coverage
*/
