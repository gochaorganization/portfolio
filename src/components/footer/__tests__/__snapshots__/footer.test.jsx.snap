// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests renders correctly 1`] = `ShallowWrapper {}`;
